src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.managed.cs

```diff
@@ -768,31 +813,34 @@ protected void AddView(FrameworkElement view, GeneratorDirection fillDirection,
 				new Point(breadthOffset, extentOffset + extentOffsetAdjustment) :
 				new Point(extentOffset + extentOffsetAdjustment, breadthOffset);
 
-			var adjustedDesiredSize = ScrollOrientation == Orientation.Vertical
-				? new Size(
-					ShouldMeasuredBreadthStretch ? AvailableBreadth :
-					(_lastMeasuredSize != default ? GetBreadth(_lastMeasuredSize) : view.DesiredSize.Width),
-					view.DesiredSize.Height
-				)
-				: new Size(
-					view.DesiredSize.Width,
-					ShouldMeasuredBreadthStretch ? AvailableBreadth :
-					(_lastMeasuredSize != default ? GetBreadth(_lastMeasuredSize) : view.DesiredSize.Height)
-				);
-
-			var finalRect = new Rect(topLeft, adjustedDesiredSize);
+			// TODO: GetElementBounds()
+			var finalRect = new Rect(topLeft, view.DesiredSize);
 
 			if (this.Log().IsEnabled(LogLevel.Debug))
 			{
-				this.Log().LogDebug($"{GetMethodTag()} finalRect={finalRect} AvailableBreadth={AvailableBreadth} adjustedDesiredSize={adjustedDesiredSize} DC={view.DataContext}");
+				this.Log().LogDebug($"{GetMethodTag()} finalRect={finalRect} AvailableBreadth={AvailableBreadth} DesiredSize={view.DesiredSize} DC={view.DataContext}");
 			}
 
-			view.Arrange(finalRect);
+			SetBounds(view, finalRect);
+		}
+
+		protected abstract Rect GetElementArrangeBounds(/*TODO ElementType, */int elementIndex, Rect containerBounds, Size windowConstraint, Size finalSize);
```

Comment:
Related ToDo Issue needed for this one ?

Reply:
This bit will be relevant once grouping+group headers are supported.